src/libraries/System.Private.Xml.Linq/src/System/Xml/Linq/Extensions.cs

```diff
@@ -311,45 +313,45 @@ public static void Remove(this IEnumerable<XAttribute> source)
         /// T which must be a derived from <see cref="XNode"/>.  Note that this method uses snapshot semantics
         /// (copies the <see cref="XNode"/>s to an array before deleting each).
         /// </summary>
-        public static void Remove<T>(this IEnumerable<T> source) where T : XNode
+        public static void Remove<T>(this IEnumerable<T?> source) where T : XNode
         {
             if (source == null) throw new ArgumentNullException(nameof(source));
 
             int count;
-            T[] nodes = EnumerableHelpers.ToArray(source, out count);
+            T?[] nodes = EnumerableHelpers.ToArray(source, out count);
             for (int i = 0; i < count; i++)
             {
-                T node = nodes[i];
+                T? node = nodes[i];
                 if (node != null) node.Remove();
             }
         }
 
-        private static IEnumerable<XAttribute> GetAttributes(IEnumerable<XElement> source, XName name)
+        private static IEnumerable<XAttribute> GetAttributes(IEnumerable<XElement?> source, XName? name)
         {
-            foreach (XElement e in source)
+            foreach (XElement? e in source)
             {
                 if (e != null)
                 {
-                    XAttribute a = e.lastAttr;
+                    XAttribute a = e.lastAttr!;
```

Comment:
nit: `XAttribute? a`, you have a null check just after anyway.

Reply:
Right.